### PR TITLE
fix(components): [scrollbar] restore scroll position in onActivated hook

### LINKED ISSUE #11255: [Component] [table] 从keepalive恢复时，表格内容的位置会重置，但横向滚动条会保持在上次的位置
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.28`
- Browser / OS: `Edge/108.0.1462.76`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBDb21wMSBmcm9tICcuL0NvbXAxLnZ1ZSdcblxuY29uc3QgdGFiID0gcmVmKCdDb21wMScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtdGFicyB2LW1vZGVsPVwidGFiXCI+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiQ29tcDFcIiBuYW1lPVwiQ29tcDFcIj48L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIkNvbXAyXCIgbmFtZT1cIkNvbXAyXCI+PC9lbC10YWItcGFuZT5cbiAgPC9lbC10YWJzPlxuICA8S2VlcEFsaXZlPlxuICAgIDxDb21wMSB2LWlmPVwidGFiID09ICdDb21wMSdcIj48L0NvbXAxPlxuICA8L0tlZXBBbGl2ZT5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXG4gIH1cbn0iLCJDb21wMS52dWUiOiI8c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBsaXN0ID0gcmVmKFt7cm93MTogJ3N0YXJ0Jywgcm93MjogJ2VuZCd9XSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxlbC10YWJsZSA6ZGF0YT1cImxpc3RcIiBib3JkZXI+XG4gICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwicm93MVwiIHByb3A9XCJyb3cxXCIgd2lkdGg9XCI1MDBcIj48L2VsLXRhYmxlLWNvbHVtbj5cbiAgICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJyb3cyXCIgcHJvcD1cInJvdzJcIiB3aWR0aD1cIjUwMFwiPjwvZWwtdGFibGUtY29sdW1uPlxuICAgIDwvZWwtdGFibGU+XG48L3RlbXBsYXRlPiIsIl9vIjp7fX0=)

### Steps to reproduce

1.拖动表格至最右侧
2.切换到另一个tab,再切换回来

### What is Expected?

表格内容和横向滚动条保持一致

### What is actually happening?

表格内容位置会重置，横向滚动条保持

### Additional comments

_(empty)_

### LINKED ISSUE #16257: [Component] [table] vue-router的RouterView使用KeepAlive后，切换router会出现el-table的滚动条和滚动位置不匹配的情况
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.21`
- Element Plus Version: `2.6.2`
- Browser / OS: `Chrome 101.0.4951.54`
- Build Tool: `Vite`

### Related Component

- `el-table`

### Reproduction Link

[CodeSandbox](https://codesandbox.io/p/devbox/test-el-table-fj684z?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu88xt6t0007356ghh95bxe2%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu88xt6s0002356g1k7z6mzg%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu88xt6s0004356gbwe7okyp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu88xt6t0006356gn2kqb95a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu88xt6s0002356g1k7z6mzg%2522%253A%257B%2522id%2522%253A%2522clu88xt6s0002356g1k7z6mzg%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clu88xt6t0006356gn2kqb95a%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu88xt6t0005356g6uuvogbg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522pnpm%2520run%2520dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252Fabout%2522%257D%255D%252C%2522id%2522%253A%2522clu88xt6t0006356gn2kqb95a%2522%252C%2522activeTabId%2522%253A%2522clu88xt6t0005356g6uuvogbg%2522%257D%252C%2522clu88xt6s0004356gbwe7okyp%2522%253A%257B%2522id%2522%253A%2522clu88xt6s0004356gbwe7okyp%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu88xt6s0003356g7i97bpwc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522pnpm%2520run%2520dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clu88xt6s0003356g7i97bpwc%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

1.点击About
2.拖动el-table的横向滚动条
3.切换到Home再切换回去About
4.可以看到滚动条和滚动位置不匹配了

### What is Expected?

滚动条和滚动位置一致

### What is actually happening?

滚动条和滚动位置不一致

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by makedopamine**
closes #11255
closes #16257 

[after](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgQ29tcDEgZnJvbSAnLi9Db21wMS52dWUnXG5cbmNvbnN0IHRhYiA9IHJlZignQ29tcDEnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGVsLXRhYnMgdi1tb2RlbD1cInRhYlwiPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIkNvbXAxXCIgbmFtZT1cIkNvbXAxXCI+PC9lbC10YWItcGFuZT5cbiAgICA8ZWwtdGFiLXBhbmUgbGFiZWw9XCJDb21wMlwiIG5hbWU9XCJDb21wMlwiPjwvZWwtdGFiLXBhbmU+XG4gIDwvZWwtdGFicz5cbiAgPEtlZXBBbGl2ZT5cbiAgICA8Q29tcDEgdi1pZj1cInRhYiA9PSAnQ29tcDEnXCI+PC9Db21wMT5cbiAgPC9LZWVwQWxpdmU+XG48L3RlbXBsYXRlPiIsInNyYy9QbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTEzNjMtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwic3JjL2VsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vcHJldmlldy0xMTM2My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcycsICdodHRwczovL3ByZXZpZXctMTEzNjMtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbChzdHlsZXMpXG59Iiwic3JjL0NvbXAxLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XHJcbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcclxuXHJcbmNvbnN0IGxpc3QgPSByZWYoW3tyb3cxOiAnc3RhcnQnLCByb3cyOiAnZW5kJ31dKVxyXG48L3NjcmlwdD5cclxuXHJcbjx0ZW1wbGF0ZT5cclxuICA8ZWwtdGFibGUgOmRhdGE9XCJsaXN0XCIgYm9yZGVyPlxyXG4gICAgICA8ZWwtdGFibGUtY29sdW1uIGxhYmVsPVwicm93MVwiIHByb3A9XCJyb3cxXCIgd2lkdGg9XCI2MDBcIj48L2VsLXRhYmxlLWNvbHVtbj5cclxuICAgICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cInJvdzJcIiBwcm9wPVwicm93MlwiIHdpZHRoPVwiNjAwXCI+PC9lbC10YWJsZS1jb2x1bW4+XHJcbiAgICA8L2VsLXRhYmxlPlxyXG48L3RlbXBsYXRlPiIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTExMzYzLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

### CODE REVIEW

File: packages/components/scrollbar/src/bar.vue [hunk 1]
```diff
@@ -29,6 +29,11 @@ const handleScroll = (wrap: HTMLDivElement) => {
   }
 }
 
+onDeactivated(() => {
+  moveX.value = 0
+  moveY.value = 0
+})
+
```
Comment (ryuhangyeong) [inline packages/components/scrollbar/src/bar.vue:36]: Thank you for your contribution!

## my opinion

When a user uses keep-alive, the user would expect to have the previous state intact. But the change doesn't seem to keep the previous state by initializing moveX, moveY to 0.

Comment (makedopamine) [reply packages/components/scrollbar/src/bar.vue:36]: You're right. I'll modify the code ASAP.

**Review by ryuhangyeong — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
@jw-foss 🙇 could you please review this PR?

**Comment by jw-foss** [comment]
> @jw-foss 🙇 could you please review this PR?

Looking 👀

**Review by jw-foss — APPROVED** [review]
@makedopamine left a comment for your consideration🏓 , let me know if you have any questions.

File: packages/components/scrollbar/src/bar.vue [hunk 2]
```diff
@@ -29,7 +35,13 @@ const handleScroll = (wrap: HTMLDivElement) => {
   }
 }
 
+const restore = (wrap: HTMLDivElement) => {
```
[the remark this answers is not in the document]
Comment (makedopamine) [reply packages/components/scrollbar/src/bar.vue:38]: If I understand right, the param of  the `handleScroll` method also should be renamed as `scrollbar` or `scrollbarEl`, is that right?🤔

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**